### ISSUE TITLE
link-fragments: no warning when heading produces an unverifiable fragment (sphinx id1/id2 fallback)

### ISSUE THREAD

**Issue by shinagawa-web** [shinagawa-web/gomarklint#223]
## Overview

When a heading slug resolves to an empty string (e.g. digits-only headings under `sphinx`, or CJK-only headings), the heading is silently excluded from the valid slug set. No diagnostic is emitted to inform the user that the fragment cannot be statically verified.

## Problem

Affected algorithms: `sphinx` (docutils `make_id()` strips leading digits and leading hyphens, so `# 123` → `""`). The same already applied to non-Latin headings (e.g. `# 日本語`).

When `slugSphinx` (or any slugger) returns `""`, `buildSlugSet` skips the entry:

```go
if base == "" {
    continue // silently dropped
}
```

A user who then writes `[text](#id1)` or `[text](#123)` receives only a generic error:

```
link-fragments: fragment #id1 not found in this document
```

There is no hint that the heading exists but its ID is assigned dynamically by Sphinx (`id1`, `id2`, …) and cannot be reproduced outside the build context.

## Discussion

Possible approaches:

1. **Emit a warning on the heading line** — when a heading's slug is `""`, report something like `link-fragments: heading "123" produces no verifiable fragment ID under sphinx (Sphinx assigns id1/id2 dynamically)`.
2. **Emit a warning on the link line** — if a fragment is not found AND the document contains at least one unverifiable heading, append a note to the existing "not found" message.
3. **Accept `idN` patterns as always-valid** — treat `#id1`, `#id2`, … as unconditionally valid under `sphinx`. Simple but masks real broken links.
4. **Do nothing** — keep current behavior; document it in the user-facing docs instead.

## References

- Fix that introduced the leading-digit case: #212
- `buildSlugSet`: `internal/rule/link_fragments_slug.go`
- Existing empty-slug comment in `slugSphinx`: `internal/rule/link_fragments_slug.go`

**Comment by shinagawa-web** [comment]
## Current status

Documenting the limitation rather than implementing a code fix for now. PR #230 adds:
- A FAQ entry explaining the false positive and two workarounds (rename the heading / use a disable comment)
- An expanded note in the `sphinx` row of the rules table

## Future plan

If this affects enough users, we will implement **Option 2**: when a fragment is not found and the document contains at least one unverifiable heading (empty slug), append a hint to the existing "not found" diagnostic pointing out that the heading's ID is assigned dynamically at build time.

Please 👍 this issue or leave a comment if you are affected — that will help us prioritise.
